model_analyzer/config/generate/quick_run_config_generator.py

```diff
@@ -122,43 +146,80 @@ def set_last_results(self, measurements):
 
-            self._coordinate_data.set_throughput(
-                coordinate=self._coordinate_to_measure, throughput=throughput)
+            self._neighborhood.coordinate_data.set_measurement(
+                coordinate=self._coordinate_to_measure, measurement=measurements[0])
             logger.debug(
-                f"Throughput for {self._coordinate_to_measure}: {self._get_last_results()}"
+                f"Measurement for {self._coordinate_to_measure}: "
+                f"throughput = {throughput}, avg_latency = {avg_latency}"
             )
         else:
-            self._coordinate_data.set_throughput(
-                coordinate=self._coordinate_to_measure, throughput=0)
-            logger.debug(f"Throughput for {self._coordinate_to_measure}: 0")
+            self._neighborhood.coordinate_data.set_measurement(
+                coordinate=self._coordinate_to_measure, measurement=None)
+            logger.debug(
+                f"Measurement for {self._coordinate_to_measure}: None."
+            )
+
+        self._update_best_measurement(measurements)
 
-    def _get_last_results(self):
-        return self._coordinate_data.get_throughput(self._coordinate_to_measure)
+    def _update_best_measurement(self, measurements: List[RunConfigMeasurement]):
+        """Keep track of the best coordinate/measurement seen so far."""
+        measurement = measurements[0]
+
+        if self._best_measurement is None:
+            self._best_measurement = measurement
+        elif measurement and measurement.is_better_than(self._best_measurement):
+            self._best_coordinate = self._coordinate_to_measure
+            self._best_measurement = measurement
+
+    def _get_last_results(self) -> RunConfigMeasurement:
+        return self._neighborhood.coordinate_data.get_measurement(
+            coordinate=self._coordinate_to_measure)
 
     def _take_step(self):
         magnitude = self._get_magnitude()
 
         new_coordinate = self._neighborhood.calculate_new_coordinate(magnitude)
         self._determine_if_done(new_coordinate)
 
-        logger.debug(f"Stepping {self._current_coordinate}->{new_coordinate}")
-        self._current_coordinate = new_coordinate
+        logger.debug(f"Stepping {self._home_coordinate}->{new_coordinate}")
+        self._home_coordinate = new_coordinate
         self._coordinate_to_measure = new_coordinate
         self._recreate_neighborhood()
 
-    def _determine_if_done(self, new_coordinate):
+    def _take_step_back(self):
+        new_coordinate = self._neighborhood.get_nearest_neighbor(
+            coordinate_in=self._best_coordinate)
+
+        logger.debug(
+            f"Stepping back: {self._home_coordinate}->{new_coordinate}"
+        )
+        self._home_coordinate = new_coordinate
+        self._coordinate_to_measure = new_coordinate
+        self._recreate_neighborhood()
+
+        self._magnitude_scaler *= MAGNITUDE_DECAY_RATE
+
+    def _is_home_measurement_none(self):
```

Comment:
the name doesn't match the functionality. This checks if we are currently measuring home and it is none.